spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/CrudMethodMetadataPostProcessor.java

```diff
@@ -119,10 +119,9 @@ static MethodInvocation currentInvocation() throws IllegalStateException {
 
 			MethodInvocation mi = currentInvocation.get();
 
-			if (mi == null)
-				throw new IllegalStateException(
-						"No MethodInvocation found: Check that an AOP invocation is in progress, and that the "
+			Assert.notNull(mi, "No MethodInvocation found: Check that an AOP invocation is in progress, and that the "
```

Comment:
This change changes behavior into throwing `IllegalArgumentException` instead of `IllegalStateException`.